### DISCUSSION TITLE
How to use SpeedTracker

### DISCUSSION THREAD

**Discussion by zj0042** [ctu-mrs/mrs_uav_system#168] [Q&A]
Hi I am trying to use SpeedTracker to go along with a directions. But not sure why the uav can not go along a line. 
Is there anybody help with this? Appreciate your help!

My configs are: 
>> uav_manager.yaml

version: "1.0.4.0"

require_gain_manager: false
require_constraint_manager: false

takeoff:

  after_takeoff:
    controller: "MpcController"
    tracker: "MpcTracker"

>>SpeedTracker command: 

rostopic pub /uav2/control_manager/speed_tracker/command mrs_msgs/SpeedTrackerCommand "header:
  seq: 0
  stamp: {secs: 0, nsecs: 0}
  frame_id: 'fcu'
velocity: {x: 0.5, y: 0.0, z: 0.0}
acceleration: {x: 0.0, y: 0.0, z: 0.0}
force: {x: 0.0, y: 0.0, z: 0.0}
height: 5.0
heading: 0.0
heading_rate: 0.0
use_velocity: true 
use_acceleration: false
use_force: false
use_height: true 
use_heading: true 
use_heading_rate: false" -r 10

Here, I used several frame_id (fcu, fcu_tilted, gps_origin), but all failed. The uav always go to x + y directions. It seems that speed on y is small.

**Comment by matemat13** [comment]
hello! Make sure that you're publishing the message periodically at a sufficient rate (at least 10Hz). For safety reasons, the speed tracker will time out after not receiving any messages for a relatively short time (I think it's something like 100ms from the top of my head).
